onnx/version_converter/adapters/axes_attribute_to_input.h

```diff
@@ -16,27 +17,28 @@ class AxesAttributeToInput : public Adapter {
   explicit AxesAttributeToInput(const std::string& op_name, const OpSetID& initial, const OpSetID& target)
       : Adapter(op_name, initial, target) {}
 
-  void attrToInput(std::shared_ptr<Graph> graph, Node* node, std::vector<int64_t> axes) const {
-    Tensor t;
-    t.elem_type() = TensorProto_DataType_INT64;
-    t.sizes() = std::vector<int64_t>{static_cast<int64_t>(axes.size())};
-    auto& data = t.int64s();
-    for (auto a : axes) {
-      data.emplace_back(a);
-    }
-
-    Node* constant = graph->create(kConstant);
-    constant->insertBefore(node);
-    constant->t_(kvalue, t);
-    node->addInput(constant->output());
+  void attrToInput(GraphProto* graph_proto, NodeProto* node_proto, std::vector<int64_t>& axes) const {
```

Comment:
Just out of curiosity: is it a convention to pass protos by pointer?

Reply:
good point! previously it was passing shared_ptr. It required the variable was a shared_ptr at calling site so it had to be always not a nullptr. To pass as reference is not good. I am using pointer, which is not as reliable. I will change it back to reference.